src/core/ipv4/dhcp.c

```diff
@@ -1476,6 +1511,20 @@ dhcp_option_hostname(u16_t options_out_len, u8_t *options, struct netif *netif)
 }
 #endif /* LWIP_NETIF_HOSTNAME */
 
+#if LWIP_DHCP_OPTION_CLIENT_IDENTIFIER
+static u16_t
+dhcp_option_client_identifier(u16_t options_out_len, u8_t *options, struct netif *netif)
+{
+  size_t i;
+  options_out_len = dhcp_option(options_out_len, options, DHCP_OPTION_CLIENT_ID, DHCP_OPTION_CLIENT_ID_MAC_LEN);
```

Comment:
Since dynamic length is used later (netif->hwaddr_len) it seems wrong to have a constant here